packages/page-template-modal/src/utils/group-utils.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * External dependencies
+ */
+import { intersection, difference } from 'lodash';
+
+/**
+ * Sorts the keys on the group object to have a preferred order.
+ * If some groups exist without a preferred order, they will be included last
+ *
+ * @param {Array<string>} preferredGroupOrder the order of group slugs that we want
+ * @param {object} groupsObject an object with all group information, with group names as keys
+ */
+export function sortGroupNames( preferredGroupOrder, groupsObject ) {
+	const groups = Object.keys( groupsObject );
+
+	const orderedGroups = intersection( preferredGroupOrder, groups );
+	const remainingGroups = difference( groups, preferredGroupOrder );
+	const allGroups = orderedGroups.concat( remainingGroups );
```

Comment:
Just thinking whether it's worth retaining some of the old behaviour and sorting the leftover groups.

```suggestion
	const allGroups = orderedGroups.concat( remainingGroups.sort() );
```

Maybe not. It means the remaining sort order is coming from the server, and that's not just a bad thing.